src/content/reference/react/captureOwnerStack.md

```diff
@@ -0,0 +1,301 @@
+
+Neither `Navigation` nor `legend` are in the stack at all since it's only a sibling to a node containing `<SubComponent />`.
+
+`SubComponent` is omitted because it's already part of the callstack.
+
+</DeepDive>
+
+## Usage {/*usage*/}
+
+### Improve error reporting {/*improve-error-reporting*/}
+
+Check out the following example to see how to use `captureOwnerStack` to improve error reporting:
+
+- [createRoot usage: Show a dialog for uncaught errors
+](/reference/react-dom/client/createRoot#show-a-dialog-for-uncaught-errors)
+- [createRoot usage: Displaying Error Boundary errors](/reference/react-dom/client/createRoot#show-a-dialog-for-uncaught-errors)
+- [createRoot usage: Displaying a dialog for recoverable errors](/reference/react-dom/client/createRoot#displaying-a-dialog-for-recoverable-errors)
+
+### Expanding error stacks {/*expanding-error-stacks*/}
+
+In addition to the <CodeStep step={1}>stack trace of the error</CodeStep> itself, you can use <CodeStep step={2}>`captureOwnerStack`</CodeStep> to append the Owner Stack.
+This can help trace the error especially when the error is caused by props. The Owner Stack helps trace the flow of props.
+
+
+```js src/index.js [[1, 8, "error.stack"], [2, 7, "captureOwnerStack()"]]
+import {captureOwnerStack} from 'react';
+import {createRoot} from 'react-dom/client';
+
+const root = createRoot(document.getElementById('root'), {
+  onUncaughtError: (error, errorInfo) => {
+    if (process.env.NODE_ENV !== 'production') {
+      const ownerStack = captureOwnerStack();
+      error.stack += ownerStack;
+    }
+
+    console.error('Uncaught', error);
+  },
+}).render(<App />);
+```
+
+<Sandpack>
+
+```js
+function useCustomHook() {
+  throw new Error('Boom!');
+}
+
+function Component() {
+  useCustomHook();
+}
+
+export default function App() {
+  return <Component />;
+}
+```
+
+```js src/index.js
+import {captureOwnerStack} from 'react';
+import {createRoot} from 'react-dom/client';
+import App from './App.js';
+import './styles.css';
+
+const root = createRoot(document.getElementById('root'), {
+  onUncaughtError: (error, errorInfo) => {
+    if (process.env.NODE_ENV !== 'production') {
+      const ownerStack = captureOwnerStack();
+      error.stack =
+        // The stack is only split because these sandboxes don't implement ignore-listing 3rd party frames via sourcemaps.
+        // A framework would ignore-list stackframes from React via sourcemaps and then you could just `error.stack += ownerStack`.
+        // To learn more about ignore-listing see https://developer.chrome.com/docs/devtools/x-google-ignore-list
+        error.stack.split('\n    at react-stack-bottom-frame')[0] + ownerStack;
```

Comment:
This should be a separate usage section like "How to collapse frames in DEV dialogs without sourcemaps"

Reply:
Or maybe Troubleshooting

Reply:
This also seems weird to me because React DevTools adds the owner stack in the stack trace? Why would you want to do this?

Reply:
Removed this section. Better leave this to React Devtools instead of polyfills.